core/src/main/java/fr/sncf/osrd/api/InfraManager.java

```diff
@@ -234,4 +245,32 @@ public FullInfra load(String infraId, String expectedVersion, DiagnosticRecorder
     public InfraCacheEntry getInfraCache(String infraId) {
         return infraCache.get(infraId);
     }
+
+    /**
+     * Get an infra given an id
+     */
+    public FullInfra getInfra(String infraId, String expectedVersion, DiagnosticRecorder diagnosticRecorder)
+            throws InfraLoadException, InterruptedException {
+        try {
+            infraCache.putIfAbsent(infraId, new InfraCacheEntry());
+            var cacheEntry = infraCache.get(infraId);
+            // download the infra for tests
+            if (loadIfMissing) {
+                return load(infraId, expectedVersion, diagnosticRecorder);
```

Comment:
can you please make a single function, and decide whether to load or throw an error inside the synchronized section?